pkg/controllers/webhook/controller.go

```diff
@@ -528,9 +528,28 @@ func (c *controller) isGlobalContextEntryReady(name string, gctxentries []*kyver
 	return false
 }
 
-func (c *controller) updatePolicyStatuses(ctx context.Context, webhookType string) error {
-	c.lock.Lock()
-	defer c.lock.Unlock()
+func checkIfPolicyIsConfiguredInWebhooks(policy kyvernov1.PolicyInterface, mExists, vExists bool) bool {
+	needsMutate := false
+	needsValidate := false
+	for _, rule := range policy.GetSpec().Rules {
+		if rule.HasMutate() || rule.HasVerifyImages() {
+			needsMutate = true
+		}
+		if rule.HasValidate() || rule.HasGenerate() || rule.HasMutateExisting() || rule.HasVerifyImages() {
+			needsValidate = true
+		}
+	}
+
+	if (needsMutate && !mExists) || (needsValidate && !vExists) {
+		return false
+	}
+
+	return true
+}
+
+func (c *controller) updatePolicyStatuses(ctx context.Context) error {
```

Comment:
Can we cover this by unit tests?

Reply:
I found the clients and listers hard to mock. I think it's enough to be covered by the unit tests above.